### ISSUE TITLE
This modules doesnt works with root compartment instead child compartment

### ISSUE THREAD

**Issue by SaurabhAhuja-Dev** [oracle-terraform-modules/terraform-oci-base#45]
terraform plan give below error for root compartment provided in tfvars file, but works fine with child compartment id--
module.bastion.oci_core_instance.bastion[0]: Refreshing state... [id=ocid1.instance.oc1.ap-mumbai-1.anrg6ljrbrnzudqcd55dlcmtirxqkc5ph6cndyusq6vbf2xb7i2pelb266ta]

Error: Invalid index

  on modules/admin/instance_principal.tf line 42, in resource "oci_identity_policy" "admin_instance_principal":
  42:   statements     = ["Allow dynamic-group ${oci_identity_dynamic_group.admin_instance_principal[0].name} to manage all-resources in compartment id ${data.oci_identity_compartments.compartments_id.compartments.0.id}"]
    |----------------
    | data.oci_identity_compartments.compartments_id.compartments is empty list of object

The given key does not identify an element in this collection value.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. 
# Please remove any sensitive information from configuration files before sharing them. 
```

### Steps to Reproduce

1. `terraform apply`

**Comment by sauraahu** [comment]
Either it should be document or code should behave properly for root compartment as well as for child together. That logic is currently missing to take of that scenario.

**Comment by hyder** [comment]
Why would you create this in the root compartment?

**Comment by sauraahu** [comment]
Bydefault, every default user gets root comparment and might tend to go with it  unless mentioned in document or error output needs to be more accurate instead of below-
data.oci_identity_compartments.compartments_id.compartments is empty list of object
More significant for non corporates which tend to directly work on root compartment, may be.
Only concern is, message to have more clear that "you have chosen root compartment for OKE clusters and its not supported with this module"

**Comment by hyder** [comment]
This is the base module, not the OKE module.

**Comment by sauraahu** [comment]
Same issue in both because OKE module uses this one.

**Comment by hyder** [comment]
This should be documentation then.

**Comment by hyder** [comment]
Will you be adding the documentation?

**Comment by sauraahu** [comment]
yes, i will update.

**Comment by SaurabhAhuja-Dev** [comment]
Sorry, was busy, This is pull request- https://github.com/oracle-terraform-modules/terraform-oci-base/pull/62

**Comment by hyder** [comment]
Resolved in #62
